app/services/scenes-collections/nodes/overlays/widget.ts

```diff
@@ -34,5 +34,9 @@ export class WidgetNode extends Node<ISchema, IContext> {
     context.sceneItem.source.replacePropertiesManager('widget', {
       widgetType: this.data.type
     });
+
+    // Make sure we don't override the url setting
+    delete this.data.settings['url'];
+    context.sceneItem.getObsInput().update(this.data.settings);
```

Comment:
I would prefer to use our API methods for that instead `getObsInput().update`

```
context.sceneItem.getSource().updateSettings(this.data.settings);
```
Because it's possible that for example that we will need to generate some event like `onSettingsUpdated`.

Reply:
Sounds good.  I copied this from another overlay node, and I think the `updateSettings` function didn't exist back then.

Reply:
We use `getObsInput` in a lot of places, but eventually we may want to clean all of those up and make it a private function.

Reply:
I'm not sure that is possible to make it a private source while we `obs-studio-node` methods that requires ObsInput as argument. For example the volmeter:
```
obsVolmeter.attach(source.getObsInput());
```

it will be possible if `obs-studio-node` will use for example name or id:
```
obsVolmeter.attach(source.id);
```

Reply:
Ah, makes sense.